Create vrn.md (#646)
vtex-apps/io-documentation: docs/en/Concepts/vrn.md

# VTEX Resource Name (VRN)

In VTEX IO, a VTEX Resource Name (VRN) is how we express a resource path.

> Most of the time, a resource is simply the URL that is being called. However, this concept is broad, and, to name a few, a resource can be a proxy, an API segment, an app endpoint, etc. That's why, to make it more tangible, in VTEX IO, we express a resource by a VRN, which structure carries information about the resource itself and the context in which it is being accessed.

Reviewer: ```suggestion
When interpreting a request, we name as *resource* the URL being called. However, this concept can be broad. To name a few, a request resource can be a proxy, an API segment, an app endpoint, etc. 

As a result, we identify and express a resource using a VRN whose structure carries essential information about the resource itself and the context in which it is being accessed.
```

Reviewer: - Separei em dois parágrafos diferentes pra não ficar muito pesado.
- Na primeira frase, adicionei o "When interpreting a request" pra dar mais contexto.
- Quebrei a segunda frase pra ficar mais fluído também, acho que as ideias ficam mais claras separadas assim.

> A VTEX Resource Name (VRN) is a resource path identifier used by the VTEX IO platform to express the resource path being called in a request. 
> 
> Most of the time, a resource is simply the URL that is being called. However, this concept is broad, and, to name a few, a resource can be a proxy, an API segment, an app endpoint, etc. That's why, to make it more tangible, in VTEX IO, we express a resource by a VRN, which structure carries information about the resource itself and the context in which it is being accessed.

Author: ```suggestion
A VTEX Resource Name (VRN) is how the VTEX IO platform expresses the resource path called in a request. 

A resource can be accessed by sending a request to its Uniform Resource Locator (URL). However, the resource concept is broad, and, to name a few, a resource can be a proxy, an API segment, an app endpoint, etc. 

Therefore, in VTEX IO, we identify and express a resource by using a VRN, which structure carries essential information about the resource itself and the context in which it is being accessed.
```

> The format of a VRN is:
> 
> `vrn:{service}:{region}:{account}:{workspace}:{path}`
> 
> Which elements can be described as:

Reviewer: ```suggestion
The format of a VRN is `vrn:{service}:{region}:{account}:{workspace}:{path}` whose elements can be described as:
```

- `service` - the infrastructure service that holds the app's information.

> - `region` - the region where the request was performed.

Reviewer: ```suggestion
- `region` - the region where the request was performed from.
```

Reviewer: Não sei se tá certo esse from, mas fiquei com essa sensação!

> - `account` - the tenant account.

Reviewer: ```suggestion
- `account` - the VTEX account responsible for performing the request.
```

Reviewer: Não sei se tá certo, mas tenant fica distante pra mim.

- `workspace` - the workspace used to perform a request.

> - `path` - the specific endpoint of a request.

Reviewer: ```suggestion
- `path` - the request specific endpoint.
```

Reviewer: Também não sei se tá 100% certo

> Example of real VRNs are:
> 
> - For a VTEX IO app: `vrn:apps:aws-us-east-1:vtex:example:/v2/apps`
> 
> - For an external service: `vrn:proxy:aws-us-east-1:vtex:example:api.visa.com/merchantlocator/v1/locator`

Reviewer: ```suggestion
Examples of real VRNs are:

- `vrn:apps:aws-us-east-1:vtex:example:/v2/apps` *(for a VTEX IO app)*
- `vrn:proxy:aws-us-east-1:vtex:example:api.visa.com/merchantlocator/v1/locator` *(for an external service)*
```

> Notice that for services outside VTEX IO, the full URL must be provided and that the `service` is simply `proxy`.

Reviewer: ```suggestion
When performing a request for external services (outside VTEX IO), remember to provide the full URL in the `path` element and also to define `proxy` in `service`.
```

Reviewer: Acho que essa info poderia estar em outro lugar, como uma recipe. Ela ensina mais sobre o momento da ação (fazer o request) do que sobre o conceito em si de VRN. Como essa outra recipe não existe, não sou contra deixar aí... mas é uma ideia.

> <div class="alert alert-warning">
> As in <code>vrn:apps:aws-us-east-1:vtex:example:/v2/apps/*</code>, a VRN may contain wildcards (<code>*</code>) to represent <strong>any</strong> value or variable. However, for security reasons, the wildcard is <strong>NOT</strong> allowed to replace the tenant of the request.
> </div>

Reviewer: ```suggestion
<div class="alert alert-warning">
As in <code>vrn:apps:aws-us-east-1:vtex:example:/v2/apps/*</code>, a VRN may contain wildcards (<code>*</code>) to represent <strong>any</strong> value or variable. However, for security reasons, the wildcard is <strong>NOT</strong> allowed to replace the VTEX account name responsible for performing the request.
</div>
```

Reviewer: Acho que essa info poderia estar em outro lugar, como uma recipe. Ela ensina mais sobre o momento da ação (fazer o request) do que sobre o conceito em si de VRN. E ela até pediria explicações mais avançadas. Eu, como usuária que estou aprendendo sobre VRN, fiquei um pouco confusa com essa ideia de adicionar o  * como variável, por exemplo. Sinto que é um conteúdo um tico mais avançado, talvez caberia estar em uma recipe mesmo. Como essa outra recipe não existe, não sou contra deixar aí... mas é uma ideia.
